python/orca/src/bigdl/orca/learn/tf2/ray_estimator.py

```diff
@@ -172,18 +172,31 @@ def fit(self, data, epochs=1, batch_size=32, verbose=1,
         )
 
         from bigdl.orca.data import SparkXShards
+        from bigdl.orca.data.tf.data import Dataset
+        from bigdl.orca.data.tf.tf2_data import TF2Dataset
         data, validation_data = maybe_dataframe_to_xshards(data, validation_data,
                                                            feature_cols, label_cols,
                                                            mode="fit",
                                                            num_workers=self.num_workers,
                                                            accept_str_col=True)
 
-        if isinstance(data, SparkXShards):
-            if data._get_class_name() == 'pandas.core.frame.DataFrame':
-                data, validation_data = process_xshards_of_pandas_dataframe(data, feature_cols,
-                                                                            label_cols,
-                                                                            validation_data, "fit")
-            ray_xshards = process_spark_xshards(data, self.num_workers)
+        if isinstance(data, SparkXShards) or isinstance(data, Dataset):
```

Comment:
why not
```python
if isinstance(data, SparkXShards):
   ...
elif isinstance(data, Dataset):
   ...
else:
   ...
```

Reply:
They have the same logic after these two types are converted to ray_xshards. Or create a new function?
```
                def zip_func(worker, this_partition_refs, that_partition_refs):
                    params["data_creator"] = make_data_creator(this_partition_refs)
                    params["validation_data_creator"] = \
                        make_data_creator(that_partition_refs)
                    return worker.step.remote(**params)

                worker_stats = ray_xshards.zip_reduce_shards_with_actors(val_ray_xshards,
                                                                         self.remote_workers,
                                                                         zip_func)
```